### ISSUE TITLE
What should "--num_examples_per_epoch" be set to?

### ISSUE THREAD

**Issue by kosonocky** [dauparas/ProteinMPNN#142]
What should "--num_examples_per_epoch" be set to for training? In the "submit_exp_020.sh" file, it is set to 1000. However, the default value in training.py is 1000000 !

submit_exp_020.sh
```
#!/bin/bash

#SBATCH -p gpu
#SBATCH --mem=128g
#SBATCH --gres=gpu:a100:1
#SBATCH -c 12
#SBATCH -t 7-00:00:00
#SBATCH --output=exp_020.out

source activate mlfold-test
python ./training.py \
           --path_for_outputs "./exp_020" \
           --path_for_training_data "path_to/pdb_2021aug02" \
           --num_examples_per_epoch 1000 \
           --save_model_every_n_epochs 50
```

training.py
```
argparser.add_argument("--num_examples_per_epoch", type=int, default=1000000, help="number of training example to load for one epoch")
```

In the supplements for the paper, it says:

> Models provided for the vanilla MPNN were trained with default flags:
> 
> v_48_002.pt - --num_neighbors 48 --backbone_noise 0.02 --num_epochs 150
> v_48_010.pt - --num_neighbors 48 --backbone_noise 0.10 --num_epochs 150
> v_48_020.pt - --num_neighbors 48 --backbone_noise 0.20 --num_epochs 150

So it seems like we should actually train with --num_examples_per_epoch at 1000000?

**Comment by kosonocky** [comment]
FYI this should almost certainly be 1,000,000

**Comment by echen1214** [comment]
when I try training with `--num_examples_per_epoch 1000000` the model spends an eternity in the pre-fetch phase of training. i haven't figured out a good way to work around this except to lower the number. with reloading every 2 epochs a full run seems quite slow on wall time.

**Comment by kosonocky** [comment]
Yep it makes it take much longer but I'm pretty sure it's correct this way. I haven't touched this in a while, so I may be wrong here, but when this param is low (1,000) then the val acc is computed over much fewer samples and so the loss curves plateau much earlier since it's not reflecting generalization but rather performance on a tiny subset. Also, the only way I was able to match the loss curves in Supp Fig S3D (in terms of loss shape and # opt steps) was to make this param the default of 1,000,000. Otherwise it plateaus much much earlier.
